daemon/state/state.go

```diff
@@ -1,46 +1,53 @@
 package state
 
 import (
+	"context"
 	"log"
 	"sync"
+	"time"
 
 	"github.com/NordSecurity/nordvpn-linux/config"
 	"github.com/NordSecurity/nordvpn-linux/daemon/pb"
 	"github.com/NordSecurity/nordvpn-linux/events"
 	"github.com/NordSecurity/nordvpn-linux/internal"
 )
 
+const notifyTimeout = 100 * time.Millisecond
+
 type subscriber struct {
-	stateChan chan<- interface{}
+	stateChan chan<- any
 	stopChan  <-chan struct{}
 }
 
-func newSubscriber(stateChan chan<- interface{}, stopChan <-chan struct{}) subscriber {
+func newSubscriber(stateChan chan<- any, stopChan <-chan struct{}) subscriber {
 	return subscriber{
 		stateChan: stateChan,
 		stopChan:  stopChan,
 	}
 }
 
 type StatePublisher struct {
-	mu          sync.Mutex
 	subscribers []subscriber
+	mu          sync.Mutex
 }
 
 func NewState() *StatePublisher {
 	return &StatePublisher{}
 }
 
-func (s *StatePublisher) notify(e interface{}) {
+func (s *StatePublisher) notify(e any) {
 	newSubs := []subscriber{}
 	for _, sub := range s.subscribers {
+		timeout, cancel := context.WithTimeout(context.Background(), notifyTimeout)
+		defer cancel()
 		select {
 		case <-sub.stopChan:
 			close(sub.stateChan)
 		case sub.stateChan <- e:
 			newSubs = append(newSubs, sub)
-		default:
+		case <-timeout.Done():
```

Comment:
Making this event sending mechanism more robust will be addressed in a separate ticket